lib/singlepass-backend/src/codegen_x64.rs

```diff
@@ -2948,6 +2949,125 @@ impl FunctionCodeGenerator<CodegenError> for X64FunctionCode {
                     ret,
                 );
             }
+            Operator::I32Extend8S => {
+                let loc =
+                    get_location_released(a, &mut self.machine, self.value_stack.pop().unwrap());
+                let ret = self.machine.acquire_locations(
+                    a,
+                    &[(WpType::I32, MachineValue::WasmStack(self.value_stack.len()))],
+                    false,
+                )[0];
+                self.value_stack.push(ret);
+
+                let tmpg1 = self.machine.acquire_temp_gpr().unwrap();
+                a.emit_mov(Size::S32, loc, Location::GPR(tmpg1));
+                a.emit_and(Size::S32, Location::Imm32(0xFF), Location::GPR(tmpg1));
```

Comment:
Do we really need to `and` the value with a corresponding bitmask first, while `mov{zx/sx}` is aware of the width?

Reply:
Good call!  It seems that we don't!